batch/batch/driver/instance.py

```diff
@@ -122,6 +126,19 @@ def adjust_free_cores_in_memory(self, delta_mcpu):
     def failed_request_count(self):
         return self._failed_request_count
 
+    async def check_is_active_and_healthy(self):
+        if self._state == 'active' and self.ip_address:
```

Comment:
So I don't think a non-active instance is "active and healthy".  I would put this check in `instance_monitoring_loop`, actually I would bake it into instances_by_last_updated:  I think `instances_by_last_updated` should actually be `active_instances_by_last_updated`.